added the Feat/workouts branch

MUTABAZI-D opened the pull request:
added the last piece of the jigsaw which is the workouts feature , so @kzngoga  kindly review it please !!!

MUTABAZI-D committed:
added workouts branch for workouts feature

MUTABAZI-D committed:
entered some data to the db.json

MUTABAZI-D committed:
Merge branch 'main' into feat/workouts

kzngoga reviewed (changes requested):
@MUTABAZI-D 

- I couldn't run the app due to this error. https://prnt.sc/rDEeDtLObuzw I think the issue is that you didn't add the **workoutsReducer** in the store file

I'll review more once this issue is fixed.

MUTABAZI-D committed:
added data in db.json file

MUTABAZI-D committed:
Merge branch 'feat/workouts' of https://github.com/MUTABAZI-D/fitness-tracker-md into feat/workouts

MUTABAZI-D committed:
added the workoutsReducer and added data to db.json

MUTABAZI-D commented:
yes master!!! now i think the workouts branch is ready and sorry for the inconvenience @kzngoga . so please review

kzngoga commented on src/components/layout/WorkoutsHeader.jsx:
```diff
@@ -0,0 +1,25 @@
+      }}
+    >
+      <Typography variant="h4" component={'h2'} color={'primary.main'}>
+        WORKOUTS
```
Use a lowercase text here. and style it to **uppercase**.

kzngoga commented on src/components/workouts/AddWorkoutModal.jsx:
```diff
@@ -0,0 +1,287 @@
+                {...register('duration', {
+                  required: 'Duration is required',
+                  min: {
+                    value: 1,
```
Change the **min value** to 10 minutes

kzngoga commented on src/components/workouts/AddWorkoutModal.jsx:
```diff
@@ -0,0 +1,287 @@
+                  required: 'Calories Burned is required',
+                  min: {
+                    value: 1,
+                    message: 'Calories burned must be at least 1 Kcal',
```
Change the **min value** to 50Kcal

kzngoga commented on src/components/workouts/AddWorkoutModal.jsx:
```diff
@@ -0,0 +1,287 @@
+                error={!!errors.caloriesBurned}
+                helperText={errors.caloriesBurned?.message}
+              />
+              <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Pls remove the **date field** you need to just pass the **today's date**.

kzngoga commented on src/components/workouts/WorkoutsTable.jsx:
```diff
@@ -0,0 +1,134 @@
+          color={'primary.contrastText'}
+          sx={{ marginLeft: 2 }}
+        >
+          WORKOUTS
```
Use lowercase text!

kzngoga reviewed (changes requested):
@MUTABAZI-D Pls check & fix these feedback issues:

- In the table you need to fix these issues https://prnt.sc/8iAYmLCrAUBG:
	- You need to show the user information instead of the user id only. (name & phone)
	- You also need to format the date (Show the date in this format ex: 27/06/2024)
	- The search should work on both **name** and **type** fields.
	- Pls show a text like **No workouts found!** when there are no workouts found or the search results returned no data. not like this: https://prnt.sc/S5GZHw4S9TAl
	- Also the **table results** need to be sortable for now sort by date in descending order. (meaning today's date should be displayed first)
- You also need to change the hover background for this **button** https://prnt.sc/LoCqzomNErYy the current background isn't visible. you can only reduce the opacity of the button to **0.8** when it's hovered on.
- Pls remove this field (https://prnt.sc/B5whmJ-stB8O) just send today's date when adding the workout. also remove it in the edit form
- Before deleting a **workout** show a modal for confirming the action, it should have 2 buttons. i.e: **proceed** & **cancel**
	- When **cancel** is clicked then close the modal
	- When **Proceed** is clicked the delete the record

MUTABAZI-D committed:
added a modal of delete ,removed the date field and formatted and other required adjustement from the feat/workouts branch

MUTABAZI-D commented:
hey master!! , i just added other info from user such as phone and name , but also corrected other requested changes . so kindly re-review it please @kzngoga

kzngoga reviewed (changes requested):
@MUTABAZI-D some few things to change, before we can merge:

- Remove the UserID in the table (https://prnt.sc/07m4MREynIi5)
- This one wasn't updated:
	- You also need to change the hover background for this button [prnt.sc/LoCqzomNErYy](https://prnt.sc/LoCqzomNErYy) the current background isn't visible. you can only reduce the opacity of the button to 0.8 when it's hovered on.

MUTABAZI-D committed:
removed userId from workouts table and reduced the opacity of Add Workout button when hovered

MUTABAZI-D commented:
made requested changes master @kzngoga . please review and merge.
